### ISSUE TITLE
Rethink the `Peer`/networking concept

### ISSUE THREAD

**Issue by sebastianst** [perun-network/perun-eth-payment#11]
Currently, `ProposeChannel` needs a `Peer` object, which is created using `Client.RegisterPeer`. It would be better if the necessary state is stored in the `Client` so that `ProposeChannel` can work with `common.Address` peer addresses. Otherwise the user has to keep track of those `Peer` instances.

Note that `RegisterPeer` also currently receives a `wire.Address` (could be `common.Address`?) and that the doc of `ProposeChannel` mentions a non-existing function `AddPeer`.

I think we also have to rethink how we handle this whole peer business when we want to offer arbitrary networking à la go-perun's `wire.Bus`.

**Comment by ggwpez** [comment]
Yes removing `RegisterPeer`+`Peer` and having the `Client` do all the logic in `ProposeChannel` is reasonable.  
Initially i thought having aliasse for peers would be good, but it becomes a burden in itself.

**Comment by matthiasgeihs** [comment]
I like this idea. We also discussed something like this [here](https://github.com/perun-network/perun-eth-payment/pull/7#discussion_r614208951).

I am wondering if that works for the listener. Does the listener accept any incoming connections, even if they come from clients that were not registered before?

**Comment by ggwpez** [comment]
>I am wondering if that works for the listener. Does the listener accept any incoming connections, even if they come from clients that were not registered before?

Yes it should.
